sale_commission_advance/models/account_move.py

```diff
@@ -0,0 +1,129 @@
+from odoo import api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    @api.model
+    def create(self, vals):
+        rec = super().create(vals)
+        if rec.has_related_sale_with_down_payment():
+            # to get correct commission amount
+            rec.recompute_lines_agents()
+        return rec
+
+    def has_related_sale_with_down_payment(self):
+        return any(
+            self.invoice_line_ids.mapped(
+                lambda x: x.sale_line_ids and x.sale_line_ids.is_downpayment
+            )
+        )
+
+
+class AccountMoveLine(models.Model):
+    _inherit = "account.move.line"
+
+    @api.depends("move_id.partner_id")
+    def _compute_agent_ids(self):
+        self.agent_ids = False  # for resetting previous agents
+        down_payment_items = self.filtered(
+            lambda x: x.move_id.partner_id
+            and x.move_id.move_type[:3] == "out"
+            and x.sale_line_ids.is_downpayment
+        )
+        for record in down_payment_items:
+            agent_ids = record._prepare_agents_vals_down_payment()
+            record.update({"agent_ids": agent_ids})
+        regular_items = self.filtered(
```

Comment:
Why computing twice?

```suggestion
        regular_items = self - down_payment_items
```